### DISCUSSION TITLE
Grapesjs Custom Asset Manager with Elfinder

### DISCUSSION THREAD

**Discussion by Smanst3r** [GrapesJS/grapesjs#5395] [Q&A] [answered]
Hello, Can someone help why selected component is incorrect? 
Im trying to connect elfinder to grapesjs and using custom event as mentioned on the official website
`editor.on('asset:custom', props => {})`
And using props select method 
`props.select(asset, false);` on elfinder select callback (e.g. click)
`props.select(asset, true);`  on elfinder getFileCallback (e.g. dblclick)
On this video incorrect model is updated (_block components added just for clarity_)
Am I missing something when the modal was hidden after the first try?
Thanks!

https://github.com/GrapesJS/grapesjs/assets/8392953/3344e050-168e-4280-964b-a25230a12085

**Comment by artf** [comment]
Can you share a reproducible demo?

**Reply by Smanst3r** [reply]
I have uploaded demo here: http://grapesjs-elfinder-sample.infinityfreeapp.com/

**Reply by artf** [reply, marked as answer]
You're binding the update from the "elFinder" only once, that's why next time you're always updating the same first component.
You have to rethink a bit your logic, eg.
```js
editor.on('asset:custom', props => {
  const selectAsset = props.select; // get always the latest `props.select`
  if (props.container.querySelector(`.cms-assets-manager`)) {
    return;
  }
  // ... then use `selectAsset(...)` instead of `props.select(...)` 
```

**Reply by Smanst3r** [reply]
Thanks, that solved the problem
